src/openfermion/utils/_meanfield.py

```diff
@@ -0,0 +1,127 @@
+operators to qubit operator using the Jordan-Wigner transmation:
+
+a^dagger_j = 0.5 (X - i Y) prod_{k = 1}^{j - 1} Z_k
+a_j = 0.5 (X + i Y) prod_{k = 1}^{j - 1} Z_k
+
+These Hamiltonians live on a square lattice which has dimensions of
+x_dimension by y_dimension. They can have periodic boundary conditions or not.
+"""
+from __future__ import absolute_import
+
+from openfermion.ops import (FermionOperator,
+                             hermitian_conjugated)
+
+
+# Function to return up-orbital index given orbital index.
+def up(index):
+    return 2 * index
+
+
+# Function to return down-orbital index given orbital index.
+def down(index):
+    return 2 * index + 1
+
+
+def meanfield_dwave(x_dimension, y_dimension, tunneling, sc_gap,
+                  periodic=True, verbose=False):
+    """Return symbolic representation of a BCS mean-field d-wave Hamiltonian.
+
+    Args:
+        x_dimension: An integer giving the number of sites in width.
+        y_dimension: An integer giving the number of sites in height.
+        tunneling: A float giving the tunneling amplitude.
+        sc_gap: A float giving the magnitude of the superconducting gap.
+        periodic: If True, add periodic boundary conditions.
+        verbose: An optional Boolean. If True, print all second quantized
+            terms.
+
+    Returns:
+        meanfield_dwave_model: An instance of the FermionOperator class.
+    """
+    # Initialize fermion operator class.
+    n_sites = x_dimension * y_dimension
+    n_spin_orbitals = 2 * n_sites
+    meanfield_dwave_model = FermionOperator((), 0.0)
+
+    # Loop through sites and add terms.
+    for site in range(n_sites):
+        # Index coupled orbitals.
+        right_neighbor = site + 1
+        bottom_neighbor = site + x_dimension
+        # Account for periodic boundaries.
+        if periodic:
+            if (x_dimension > 2) and ((site + 1) % x_dimension == 0):
+                right_neighbor -= x_dimension
+            if (y_dimension > 2) and (site + x_dimension + 1 > n_sites):
+                bottom_neighbor -= x_dimension * y_dimension
+
+        # Add transition to neighbor on right
+        if (site + 1) % x_dimension or (periodic and x_dimension > 2):
+            # Add spin-up hopping term.
+            operators = ((up(site), 1), (up(right_neighbor), 0))
+            hopping_term = FermionOperator(operators, -tunneling)
+            meanfield_dwave_model += hopping_term
+            meanfield_dwave_model += hermitian_conjugated(hopping_term)
+            # Add spin-down hopping term
+            operators = ((down(site), 1), (down(right_neighbor), 0))
+            hopping_term = FermionOperator(operators, -tunneling)
+            meanfield_dwave_model += hopping_term
+            meanfield_dwave_model += hermitian_conjugated(hopping_term)
+
+            # Add pairing term
+            operators = ((up(site), 1), (down(right_neighbor), 1))
+            pairing_term = FermionOperator(operators, sc_gap / 2)
```

Comment:
What you write here is totally functional but a nit picky comment is that in python it is best to always write, e.g. "2." instead of "2" when you intend a float. Since sc_gap is a float this is okay. But if by some chance somebody provided an int like "7" instead of "7." for the sc_gap parameter, then 7/2 = 3 instead of 3.5 in some versions of python. This comment applies to a few things in this PR.